examples/pipeline-demo/main.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import re
+import random
+
+from google.cloud import vision
+from google.cloud import vision_v1
+
+from past.builtins import unicode
+
+from firebase_admin import firestore
+
+
+import apache_beam as beam
+from apache_beam.io import ReadFromText
+from apache_beam.io import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+import flickrapi
+
+photos = ['https://live.staticflickr.com/7210/6843831417_861d6996e8_c.jpg']
+
+def initializeDB(): 
+    # Use the application default credentials 
+    if not firebase_admin._apps:
+        firebase_admin.initialize_app(credentials.ApplicationDefault(), {
+        'projectId': 'step-project-ellispis',
+        })
+    return firestore.client()
+
+def callFlicker(pageNumber):
+    flickr = flickrapi.FlickrAPI('2d00397e012c30ccc33ca4fdc05a5c98', 'e36a277c77f09fdd', cache=True)
+    photos = flickr.photos.search(text='plasticbag',
+                     tag_mode='all',
+                     tags='plasticbag',
+                     extras='url_c, geo, date_upload, date_taken, owner_name, icon_server',
+                     per_page=10,  # may be you can try different numbers..
+                     page=pageNumber,
+                     sort='relevance')
+    return photos[0]
+
+def getNumOfBatches():
+    photos=callFlicker(1)
+    return (photos.attrib['pages'])
+
+class getUrl(beam.DoFn):
+    def process(self, element):
+        return [(getRandomKey(), element.get('url_c'))]
+
+def getRandomKey():
+        return random.randint(1, 10)
+
+class getLabelsByBatch(beam.DoFn):
+    def process(self, element):
+        client = vision_v1.ImageAnnotatorClient()
```

Comment:
Don't create the database connection once for each element, do it only once using `setup`:
https://beam.apache.org/releases/pydoc/2.24.0/apache_beam.transforms.core.html#apache_beam.transforms.core.DoFn.setup

Reply:
Done.